File: backoffice/src/diningTable/domain/exceptions/DiningTableNotFoundException.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain            import DomainException
from ..valueObjects.DiningTableId import DiningTableId

"""
 *
 * Classes 
 *
"""

class DiningTableNotFoundException( DomainException ):

    """
     *
     * Constants 
     *
    """

    __DINING_TABLE_NOT_FOUND = 148

    """
     *
     * Methods 
     *
    """

    def __init__( self, id : DiningTableId ) -> None:
        super().__init__( 
            self.__DINING_TABLE_NOT_FOUND,
            'The dining table {} has not been found'.format( id.getValue() )
        )

File: preparation/src/diningTable/domain/exceptions/DiningTableNotFoundException.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class DiningTableNotFoundException : DomainException {

        /* 
         * 
         * Constants
         *
        */

        private const int DINING_TABLE_NOT_FOUND = 148;

        /* 
         * 
         * Methods
         *
        */

        public DiningTableNotFoundException( DiningTableId id ) : base( 
            DINING_TABLE_NOT_FOUND,
            string.Format( "The dining table {0} has not been found", id.getValue() )
        ) {
        }

    }

}